### ISSUE TITLE
Volume Downsample Unexpected Behavior

### ISSUE THREAD

**Issue by j-c-c** [ComputationalCryoEM/ASPIRE-Python#1155]
After some investigation into our 3D downsample I'm seeing some unexpected behavior regarding centering and scaling.

Here is a small example script I'm running:

```python
import numpy as np

from aspire.volume import Volume

SIZES = [64, 65]
DS_SIZES = [32, 33]
dtype = np.float64

for res in SIZES:
    one_hot = np.zeros((res,) * 3, dtype=dtype)
    one_hot[res//2, res//2, res//2] = 1
    vol = Volume(one_hot)

    # Check vol.downsample()                                                                                                                                                                                                              
    for res_ds in DS_SIZES:
        vol_ds = vol.downsample(res_ds)

        # Look at center                                                                                                                                                                                                                  
        cent_vals = vol_ds.asnumpy()[0][
            res_ds//2 - 1: res_ds//2 + 2,
            res_ds//2 - 1: res_ds//2 + 2,
            res_ds//2 - 1: res_ds//2 + 2,
        ]
        print(f"res:{res}, res_ds:{res_ds}\n")
        print(f"{cent_vals}\n\n")
```

and the output:

```
res:64, res_ds:32

[[[0.    0.    0.   ]
  [0.    0.    0.   ]
  [0.    0.    0.   ]]

 [[0.    0.    0.   ]
  [0.    0.125 0.   ]
  [0.    0.    0.   ]]

 [[0.    0.    0.   ]
  [0.    0.    0.   ]
  [0.    0.    0.   ]]]


res:64, res_ds:33

[[[-0.00132346  0.00395838  0.00395838]
  [ 0.00395838 -0.0118393  -0.0118393 ]
  [ 0.00395838 -0.0118393  -0.0118393 ]]

 [[ 0.00395838 -0.0118393  -0.0118393 ]
  [-0.0118393   0.03541068  0.03541068]
  [-0.0118393   0.03541068  0.03541068]]

 [[ 0.00395838 -0.0118393  -0.0118393 ]
  [-0.0118393   0.03541068  0.03541068]
  [-0.0118393   0.03541068  0.03541068]]]


res:65, res_ds:32

[[[ 0.00299453  0.0093219  -0.00185942]
  [ 0.0093219   0.02873575 -0.00567675]
  [-0.00185942 -0.00567675  0.00111061]]

 [[ 0.0093219   0.02873575 -0.00567675]
  [ 0.02873575  0.08772943 -0.01716358]
  [-0.00567675 -0.01716358  0.0033247 ]]

 [[-0.00185942 -0.00567675  0.00111061]
  [-0.00567675 -0.01716358  0.0033247 ]
  [ 0.00111061  0.0033247  -0.00063736]]]


res:65, res_ds:33

[[[-0.00074854  0.00378095  0.00123554]
  [ 0.00378095 -0.01909793 -0.00624084]
  [ 0.00123554 -0.00624084 -0.00203939]]

 [[ 0.00378095 -0.01909793 -0.00624084]
  [-0.01909793  0.09646548  0.0315231 ]
  [-0.00624084  0.0315231   0.01030116]]

 [[ 0.00123554 -0.00624084 -0.00203939]
  [-0.00624084  0.0315231   0.01030116]
  [-0.00203939  0.01030116  0.00336622]]]
```

My expectation is that all of the pixel intensity would be in the center pixel and have value 1.

**Comment by janden** [comment]
Thanks for looking into this.

**Comment by j-c-c** [comment]
A little update here. 

I made a script that compares the operations `vol.downsample().project()` with `vol.project.downsample` by taking the RMSE between the images produced from each transformation. This was done using a one-hot volume and emdb_2660.

 Here are the results with plots of the images:

One-Hot Volume:
```
 Downsample/Project results:
size:8, ds_size:6, RMSE:0.006477782779331849
size:8, ds_size:7, RMSE:0.01482429097713289
size:9, ds_size:6, RMSE:0.006445406164266973
size:9, ds_size:7, RMSE:0.0026874260831369922
```

![Screenshot 2024-07-25 at 4 23 19 PM](https://github.com/user-attachments/assets/08d47a3b-3a39-43eb-b21a-0efcceedaf68)

EMDB 2660 Volume:
```
 Downsample/Project results:
size:64, ds_size:32, RMSE:0.0001832545611463177
size:64, ds_size:33, RMSE:0.0018592582536780323
size:65, ds_size:32, RMSE:0.0009730553655656341
size:65, ds_size:33, RMSE:0.0009615278087774933
```

![Screenshot 2024-07-25 at 4 26 51 PM](https://github.com/user-attachments/assets/2820e1eb-5ba0-4308-9c32-f57fbe390f14)

Follow up comment to discuss the results from the changes in PR #1162

**Comment by j-c-c** [comment]
With the changes added in PR #1162 here are the new results:

One-Hot Volume:
```
 Downsample/Project results:
size:8, ds_size:6, RMSE:1.973389292427291e-11
size:8, ds_size:7, RMSE:4.471971839125424e-11
size:9, ds_size:6, RMSE:2.1375030531204312e-17
size:9, ds_size:7, RMSE:2.6315002537815024e-11
```

![Screenshot 2024-07-25 at 4 31 30 PM](https://github.com/user-attachments/assets/6b520735-a807-41f0-9b87-681067cd2943)

EMDB 2660 Volume:
```
 Downsample/Project results:
size:64, ds_size:32, RMSE:1.5354757327312151e-12
size:64, ds_size:33, RMSE:2.4953439475934754e-11
size:65, ds_size:32, RMSE:2.335317233600185e-11
size:65, ds_size:33, RMSE:1.6181744639067254e-11
```

![Screenshot 2024-07-25 at 4 33 11 PM](https://github.com/user-attachments/assets/451680ff-4695-4492-b9a1-c3f5996b25cd)

**Comment by j-c-c** [comment]
Closing. Resolved in #1162
